pupil_src/shared_modules/annotations.py

```diff
@@ -7,239 +7,329 @@
+
+        self._append_button_quickbar(annotation_info)
+        self._append_button_menu(annotation_info)
+
+    def _create_button_quickbar(self, annotation_definition):
+        def make_fire(definition):
+            return lambda _: self.fire_annotation(definition)
+
+        return ui.Thumb(
+            annotation_definition.name,
+            setter=make_fire(annotation_definition),
+            getter=lambda: False,
+            label=annotation_definition.hotkey,
+            hotkey=annotation_definition.hotkey,
+        )
+
+    def _create_button_menu(self, annotation_info):
+        def make_remove(info):
+            return lambda: self._remove_annotation_info(info)
+
+        name = annotation_info.definition.name
+        hotkey = annotation_info.definition.hotkey
+        return ui.Button(
+            label=name + " <" + hotkey + ">", function=make_remove(annotation_info)
+        )
+
+    def _append_button_quickbar(self, annotation_info):
+        current_buttons = self.g_pool.quickbar.elements
+        index_of_last_button = -1
+        for other_annotation in self._annotation_infos:
+            if not other_annotation.button_quickbar:
+                continue
+            try:
+                idx = current_buttons.index(other_annotation.button_quickbar)
+            except ValueError:
+                pass
+            else:
+                index_of_last_button = max(idx, index_of_last_button)
+        if index_of_last_button != -1:
+            self.g_pool.quickbar.insert(
+                index_of_last_button + 1, annotation_info.button_quickbar
+            )
+        else:
+            self.g_pool.quickbar.append(annotation_info.button_quickbar)
 
-    def on_notify(self, notification):
-        if notification['subject'] == 'annotation':
-            logger.info('Received {} annotation'.format(notification['label']))
+    def _append_button_menu(self, annotation_info):
+        self._annotation_list_menu.append(annotation_info.button_menu)
 
-    def get_init_dict(self):
-        return {'annotation_definitions': self.annotation_definitions}
+    def _remove_annotation_info(self, annotation_info):
+        self._remove_button_quickbar(annotation_info)
+        self._remove_button_menu(annotation_info)
+        self._annotation_infos.remove(annotation_info)
+
+    def _remove_button_quickbar(self, annotation_info):
+        if annotation_info.button_quickbar:
+            self.g_pool.quickbar.remove(annotation_info.button_quickbar)
+            annotation_info.button_quickbar = None
 
+    def _remove_button_menu(self, annotation_info):
+        if annotation_info.button_menu:
+            self._annotation_list_menu.remove(annotation_info.button_menu)
+            annotation_info.button_menu = None
 
-class Annotation_Player(Annotation_Capture, Analysis_Plugin_Base):
-    """Describe your plugin here
-    View,edit and add Annotations.
+
+def create_annotation(label, timestamp):
```

Comment:
Is there a reason why this is not a static method of `AnnotationPlugin`?

Reply:
I like the idea